### PR TITLE
Fix names of the Bolt driver metrics

### PR DESCRIPTION
**Pull request by NataliaIvakina**
Separate Bolt driver metrics from the Bolt metrics.
Fix names of the Bolt driver metrics according to the codebase.
For details, see https://github.com/neo-technology/neo4j/pull/22339. These metrics were added as a new metric class.

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1385.surge.sh

Updated pages:
- [Metrics reference](https://neo4j-docs-operations-1385.surge.sh/operations-manual/5/monitoring/metrics/reference)

File: modules/ROOT/pages/monitoring/metrics/reference.adoc [hunk 1]
```diff
@@ -92,10 +92,18 @@ By default, database metrics include:
...
 |<prefix>.bolt.response_ignored|(unsupported feature) When `internal.server.bolt.response_metrics` is enabled, number of `encounteredignored` responses (counter)
 |<prefix>.bolt.response_failed|(unsupported feature) When `internal.server.bolt.response_metrics` is enabled, number of `encounteredinstances` of a given error code. (counter)
-|<prefix>.bolt_driver.managed_transaction_function_calls|The total number of managed transaction function calls. (counter)
-|<prefix>.bolt_driver.unmanaged_transaction_calls|The total number of un-managed transaction function calls. (counter)
-|<prefix>.bolt_driver.implicit_transaction_calls|The total number of implicit transaction function calls. (counter)
-|<prefix>.bolt_driver.execute_calls|The total number of driver-level execute function calls. (counter)
+|===
+
+[[bolt-driver-metrics]]
+.Bolt Driver metrics
```
Comment (renetapopova) [inline modules/ROOT/pages/monitoring/metrics/reference.adoc:98]:
```suggestion
.Bolt driver metrics
```

Comment (renetapopova) [reply modules/ROOT/pages/monitoring/metrics/reference.adoc:98]: I wonder if we could have labels in the description. If that's not possible, we can add it above the table as we do it for the configuration settings.

File: modules/ROOT/pages/monitoring/metrics/reference.adoc [hunk 2]
```diff
@@ -92,10 +92,18 @@ By default, database metrics include:
...
+
+[options="header",cols="<3m,<4"]
+|===
+|Name |Description
+|<prefix>.dbms.bolt_driver.api.managed_transaction_function_calls|label:new[Introduced in 5.13]The total number of managed transaction function calls. (counter)
+|<prefix>.dbms.bolt_driver.api.unmanaged_transaction_calls|label:new[Introduced in 5.13]The total number of un-managed transaction function calls. (counter)
```
Comment (renetapopova) [inline modules/ROOT/pages/monitoring/metrics/reference.adoc:104]:
```suggestion
|<prefix>.dbms.bolt_driver.api.unmanaged_transaction_calls|label:new[Introduced in 5.13]The total number of unmanaged transaction function calls. (counter)
```

**Review by renetapopova — COMMENTED** [review]
Thanks, @NataliaIvakina. I added some suggestions, but feel free to ignore them.

File: modules/ROOT/pages/monitoring/metrics/reference.adoc (hunk 1 above)
Comment (NataliaIvakina) [reply modules/ROOT/pages/monitoring/metrics/reference.adoc:98]: 'Driver` starts with capital letter in the code. I'd prefer to keep spelling as it's in the code, we can fix it later.

Comment (NataliaIvakina) [reply modules/ROOT/pages/monitoring/metrics/reference.adoc:98]: I like the idea about labels, it would be much better to have one label instead of four.

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1385.surge.sh

Updated pages:
- [Metrics reference](https://neo4j-docs-operations-1385.surge.sh/operations-manual/5/monitoring/metrics/reference)

**Review by renetapopova — APPROVED** [review]

**Pull request merged**